documentation/specs/src/interoperability/ethereum-bridge/ethereum_events_attestation.md

```diff
@@ -54,104 +54,93 @@ keys involved are:
+pub struct Vext {
+    /// The block height for which this [`Vext`] was made.
+    pub block_height: BlockHeight,
     /// The address of the signing validator
-    signer: Address,
-    /// The proportion of the total voting power held by the validator
-    power: FractionalVotingPower,
-    /// The event being signed and the block height at which
-    /// it was seen. We include the height as part of enforcing
-    /// that a block proposer submits vote extensions from
-    /// **the previous round only**
-    event: Signed<(EthereumEvent, BlockHeight)>,
+    pub validator_addr: Address,
+    /// The new ethereum events seen. These should be
+    /// deterministically ordered.
+    pub ethereum_events: Vec<EthereumEvent>,
 }
 ```
 
-These vote extensions will be given to the next block proposer who will
-aggregate those that it can verify and will inject a protocol transaction
-(the "vote extensions" transaction).
-
-```rust
-pub struct MultiSigned<T: BorshSerialize + BorshDeserialize> {
-    /// Arbitrary data to be signed
-    pub data: T,
-    /// The signature of the data
-    pub sigs: Vec<common::Signature>,
-}
-
-pub struct MultiSignedEthEvent {
-    /// Address and voting power of the signing validators
-    pub signers: Vec<(Address, FractionalVotingPower)>,
-    /// Events as signed by validators
-    pub event: MultiSigned<(EthereumEvent, BlockHeight)>,
-}
-
-pub enum ProtocolTxType {
-    EthereumEvents(Vec<MultiSignedEthEvent>)
-}
-```
+These votes will be given to the next block proposer who will
+aggregate those that it can verify and will inject a signed protocol
+transaction into their proposal.
 
-This vote extensions transaction will be signed by the block proposer. 
-Validators will check this transaction and the validity of the new votes as 
+Validators will check this transaction and the validity of the new votes as
 part of `ProcessProposal`, this includes checking:
 - signatures
 - that votes are really from active validators
 - the calculation of backed voting power
 
-It is also checked that each vote extension came from the previous round, 
-requiring validators to sign over the Namada block height with their vote
-extension. Furthermore, the vote extensions included by the block proposer
-should have at least 2 / 3 of the total voting power of the previous round 
-backing it. Otherwise the block proposer would not have passed the 
-`FinalizeBlock` phase of the last round. These checks are to prevent censorship 
-of events from validators by the block proposer.
+If vote extensions are supported, it is also checked that each vote extension
+came from the previous round, requiring validators to sign over the Namada block
+height with their vote extension.
+
+Furthermore, the vote extensions included by
+the block proposer should have at least 2 / 3 of the total voting power of the
+previous round backing it. Otherwise the block proposer would not have passed the
+`FinalizeBlock` phase of the last round.
```

Comment:
`phase of the last round of the last block.`

sorry I know I'm being a pain in the ass lol